kf_lib_data_ingest/etl/transform/guided.py

```diff
@@ -198,3 +205,44 @@ def _do_transform(self, data_dict):
         target_instances = self._standard_to_target(self.transform_func_df)
 
         return target_instances
+
+    def _postrun_concept_discovery(self, run_output):
+        """
+        See the docstring for IngestStage._postrun_concept_discovery
+        """
+        sources = defaultdict(dict)
+        links = defaultdict(
+            lambda: defaultdict(set)
+        )
+        values = defaultdict(
+            lambda: defaultdict(set)
+        )
+
+        df = self.transform_func_df
+
+        for key in df.columns:
+            sk = sources[key]
+            for val in df[key]:
+                # sources entry
+                sk[val] = True
+
+        for _, row in df.iterrows():
+            for keyA in df.columns:
+                vA = row[keyA]
+                for keyB in df.columns:
+                    if keyB != keyA:
+                        vB = row[keyB]
+                        if vA and vB:
+                            # links entry
+                            links[keyA + '::' + keyB][vA].add(vB)
+
+                            # values entry
```

Comment:
Maybe best to explain in person, but I'm not sure I understand what this is doing? Maybe for relationship counting?

Reply:
in preparation for addressing #232